### PR TITLE
[skip-ci] TMT: Reorg and reuse upstream tests on downstream updates

### PR DESCRIPTION
**Pull request by lsm5**
This commit accounts for breaking changes in `dnf5 config-manager`.

Merge after https://github.com/containers/aardvark-dns/pull/461

### CODE REVIEW

**Comment by lsm5** [comment]
@mheon @Luap99 PTAL

File: plans/main.fmf
Comment (Luap99) [inline plans/main.fmf:55]: Is there no if else syntax? Like this it could be possible to miss a distro accidentally, not that I think it is the case here just looking to make this more robust.

Comment (lsm5) [reply plans/main.fmf:55]: let me check if i can simplify this further. Putting this back to draft.

Comment (lsm5) [reply plans/main.fmf]: @Luap99 simplified a lot by using generic steps to work across all distros. There's still some conditionalizing for enabling centos-stream copr targets on centos-stream test environments and that can't be generalized unless there's changes in how dnf / copr enables targets on rhel/centos-stream envs.

File: plans/main.fmf
Comment (Luap99) [inline plans/main.fmf:9]: Is there a strong reason to do this? I rather use the packaged version. Is the issue that there is no epel 10 yet?

Comment (edsantiago) [reply plans/main.fmf:9]: I'm not sure what you mean by "`parallel` support". As best I understand, plain BATS will use GNU parallel just fine. What I see in your link above is that the bats rpm has a `Requires: parallel` -- that's something we could do ourselves.

For now and the immediate future this is moot, since system tests can't yet run in parallel.

Comment (lsm5) [reply plans/main.fmf:9]: > I'm not sure what you mean by "`parallel` support". As best I understand, plain BATS will use GNU parallel just fine. What I see in your link above is that the bats rpm has a `Requires: parallel` -- that's something we could do ourselves.
> 

@edsantiago ack, I just saw `BuildRequires: parallel` in there so I thought `support`.

> For now and the immediate future this is moot, since system tests can't yet run in parallel.

In that case I would prefer what I currently have on the PR, mainly for simplicity.

Comment (Luap99) [reply plans/main.fmf:9]: FYI we are in aardvark-dns here, we do not have parallel runs enabled but the tests are designed with sufficient isolation that it could be easily be enabled I think. 
Test timing is not that critical on this repo but I think losing this ability is bad.

Reasons why I don't like this:
- nobody ever will remember to update this there as such we keep using outdated bats forever and then don't notice if things break
- there is no trust chain, with the rpm package we get gpg signatures, sure this likely does not matter as upstream must be trusted no matter what
- there is no way to know what we get really with executing the install.sh script? What if it fails for some reason? We know have to debug this thing with no knowledge what it executes.

Comment (edsantiago) [reply plans/main.fmf:9]: FWIW, this (fetch & unpack static tarball from github) is the method used in RHEL gating tests. It is unfortunately more reliable than epel or anything else I could think of, particularly on unreleased RHELs.

Comment (lsm5) [reply plans/main.fmf:9]: 1 possible alternative I could try would be `dnf install $BATS_RPM_URL $PARALLEL_RPM_URL` where the URLs are generic enough that they'll work for all distros without any extra setup. I'll have to pull in parallel too because of the install requirement. I can check on that

**Comment by lsm5** [comment]
I'll probably end up reworking this to also account for downstream reusability.

**Comment by lsm5** [comment]
@Luap99 @edsantiago @cevich PTAL at the current state. With this setup, we can reuse the tests for Fedora zuul (before dist-git PR merge) and bodhi updates and also on CentOS Stream.

See test downstream PR: https://src.fedoraproject.org/rpms/aardvark-dns/pull-request/20 and the actual TMT test trigger in Zuul: https://artifacts.dev.testing-farm.io/34dd8354-1a3d-4d34-ba58-9fb432581b8b/ . 

Unit, validate and integration tests are run upstream while only the integration tests are run downstream.

Once everything looks ok, I'll re-enable failure notifications and repush for final merge.

File: test/tmt/tests.sh
Comment (cevich) [inline test/tmt/tests.sh:39]:
```suggestion
if [[ -n $RHEL_VERSION ]]; then
```

File: test/tmt/tests.sh
Comment (cevich) [inline test/tmt/tests.sh:48]: I think this output would be more useful to have at the start of the script.  If something above goes south, execution will never reach here.

**Review by cevich — CHANGES_REQUESTED** [review]

File: test/tmt/main.fmf
Comment (edsantiago) [inline test/tmt/main.fmf:24]: This and line 31 are trailing whitespace. I'm a little alarmed that the git hooks didn't block it.

**Review by edsantiago — APPROVED** [review]
Tentative LGTM but I'm not qualified to read packit

File: .packit.yaml
Comment (lsm5) [reply .packit.yaml:42]: yes

File: test/tmt/main.fmf
Comment (cevich) [reply test/tmt/main.fmf:24]: If you're talking about pre-commit hooks, I don't see any setup on this repo :disappointed:

File: test/tmt/tests.sh
Comment (Luap99) [inline test/tmt/tests.sh:1]: should this file be named test_integration.sh instead? I find tests.sh a weird name given it only runs integration tests so the purpose is not quite clear.

Comment (lsm5) [reply test/tmt/tests.sh:1]: sgtm

File: test/tmt/tests.sh
Comment (Luap99) [inline test/tmt/tests.sh:6]: usage script name does not match

File: test/tmt/main.fmf
Comment (Luap99) [inline test/tmt/main.fmf:9]: Why only on aarch?

File: test/tmt/tests.sh
Comment (cevich) [inline test/tmt/tests.sh:6]:
```suggestion
    echo -e "Usage: $(basename ${BASH_SOURCE[0]}) STREAM\nSTREAM can be upstream or downstream"
```

File: test/tmt/main.fmf
Comment (Luap99) [inline test/tmt/main.fmf:28]: The duplication between upstream and downstream confuses me here? Also the names make no sense as this runs integration tests not upstream/downstream specific tests. 
Can't we just assign two tags upstream and downstream to one definition?

File: test/tmt/main.fmf
Comment (lsm5) [reply test/tmt/main.fmf:9]: IIRC, the default is 5m and that sufficed for x86_64 consistently. Only aarch64 timed out sometimes.

File: test/tmt/main.fmf
Comment (lsm5) [reply test/tmt/main.fmf]: I pushed a test change. Let me see how I can best address this. Hold off on this for a bit.

File: test/tmt/main.fmf
Comment (lsm5) [reply test/tmt/main.fmf]: removed

File: .packit.yaml
Comment (lsm5) [reply .packit.yaml:23]: added comment

File: test/tmt/main.fmf
Comment (lsm5) [reply test/tmt/main.fmf]: see [this change](https://github.com/containers/aardvark-dns/pull/465/files#diff-68ada78e2ae75611599170f255698e6e31c66e0bfe3aae153e9c41b6cb7ad89fR21-R30) which reorgs the integration tests. The results are [here](https://artifacts.dev.testing-farm.io/abce20c4-0a57-4e57-9808-82822c3e9cfa/)

Given aardvark-dns tests depend on netavark, for upstream we'll need to fetch netavark from podman-next while for downstream we'll need to fetch official fedora netavark. So, there will be some differences. We `could` move this environment detection in the script itself, but hope we're cool with doing that in a followup PR.

Comment (Luap99) [reply test/tmt/main.fmf]: yes that looks good enough to me

**Comment by lsm5** [comment]
PTAL again. Good for hopefully final review and merge. Thanks!

File: test/tmt/tests.sh
Comment (lsm5) [reply test/tmt/tests.sh]: done

**Review by Luap99 — APPROVED** [review]
LGTM, I have no way to verify any of the downstream changes but looks like the upstream changes work and it is tested properly

**Comment by lsm5** [comment]
> LGTM, I have no way to verify any of the downstream changes but looks like the upstream changes work and it is tested properly

Thanks @Luap99.

If anyone would like to verify how it looks like downstream (currently a manual sync), go to the dist-git test PR: https://src.fedoraproject.org/rpms/aardvark-dns/pull-request/20# -> `Zuul` CI result -> `rpm-tmt-test` -> `Artifacts` tab -> `Testing Farm Artifacts`. Only the downstream integration tests are run there.

The automatic syncing from Packit and bodhi results will only be found after an upstream release, unless we do a manual sync first. But Zuul results should be a strong enough indicator. HTH.

**Comment by Luap99** [comment]
We cut a new release tomorrow should we get this merged for that?

**Comment by lsm5** [comment]
> We cut a new release tomorrow should we get this merged for that?

That would be great.  Thanks @Luap99 . I have further simplifications lined up for container-selinux and netavark, which I can replicate in av as well for a followup.

**Pull request merged**
